core/src/main/java/com/nuodb/migrator/jdbc/metadata/inspector/NuoDBColumnInspector.java

```diff
@@ -126,6 +127,22 @@ protected void processResultSet(InspectionContext inspectionContext, ResultSet c
                 Sequence sequence = new Sequence(identifier);
                 column.setSequence(sequence);
                 column.getTable().getSchema().addSequence(sequence);
+
+                // setting generatedAlways on dump job
+                String runningJob = getInstance().getState();
+
+                int flags = columns.getInt("FLAGS");
+                // System.out.println("flags for sequence: " + flags);
+                if (runningJob == "com.nuodb.migrator.dump.DumpJob") {
+                    if (flags >= 9) { /// flags | GENERATED_ALWAYS (which is
```

Comment:
I don't know what GENERATED_ALWAYS is referencing. I constant somewhere? The comment makes it sounds like GENERATED_ALWAYS is the 4th bit in the bit map. You should create a constant for each of these and check them using bitwise operations, e.g.
```
public static final int GENERATED_ALWAYS = 8;
...

if (flags & GENERATED_ALWAYS != 0) {...}
```

Reply:
The comment says "flags or GENERATED_ALWAYS is initialized as 8" But, GENERATED_ALWAYS is kind of just an elaboration of flag variable. So, I would change the comment to a meaningful one.

Reply:
Flags are often implemented with bitwise operations. For example, in a single byte there are 8 bits, supporting up to 8 different flag values. Each flag has a 1 set at a different binary digit in the byte.

|Flag|Binary digits|Decimal number|Written as bit shift|
|---|---|---|---|
|None|00000000|0|0|
|FirstFlag|00000001|1|1 << 0|
|SecondFlag|00000010|2|1 << 1|
|ThirdFlag|00000100|4|1 << 2|
|FourthFlag|00001000|8|1 << 3|
|...|...|...|...|

The reason we would represent flag values this way is so that they can be arbitrarily combined with bitwise operations. That wouldn't be true if we just enumerated different flag values as decimal 1, decimal 2, decimal 3, etc. For example, using the flag values defined above:

```
FirstFlag | ThirdFlag == (1 << 0) | (1 << 2) == 5 // binary digits 00000101
```

In the table you are working with here, the flag column gets its values from this enumeration using powers of 2 just like in the example above:

`Omega/Metadata/FieldDescriptor.h`
```
enum class FieldFlags : int
{
    NotNull         = 1 << 0,
    GeneratedAlways = 1 << 3,
    // Visible Computed columns
    ComputedVisible  = 1 << 4,
};
```

I don't think the code you wrote does what you want, but if it does then you still need to rewrite it to be more clear.

First let me rewrite the FieldFlags in a table so you can see the decimal values:

|Flag|Bit shift|Decimal value|
|---|---|---|
|NotNull|1 << 0|1|
|GeneratedAlways|1 << 3|8|
|ComputedVisible|1 << 4|16|

So the expression in your if-statement will evaluate to true if FieldFlags has the ComputedVisible bit set. Also in the future if any new Field Flags are added with a decimal value larger than 8, the statement will also evaluate to true for values with that flag set.

I don't think this is what you mean.

Inside the body of the if-statement you do `setGeneratedAlways(true)`. Are the following two code blocks accurate to what you are wanting to achieve with this code?

```
// Values copied from Omega/Metadata/FieldDescriptor.h
public class FieldFlags {
    public static final int NotNull = (1 << 0);
    public static final int GeneratedAlways = (1 << 3);
    public static final int ComputedVisible = (1 << 4);
}
```

```
if ((flags & FieldFlags.GeneratedAlways) != 0) {
    column.getSequence().setGeneratedAlways(true);
}
```

Also, while I don't generally approve of spending time trying to reduce number of lines, it might be more readable to get rid of the if-statement entirely:

```
column.getSequence().setGeneratedAlways((flags & FieldFlags.GeneratedAlways) != 0);
```

If you put a little refactoring work in you could probably rewrite that line to look like this:

```
column.getSequence().setGeneratedAlways(flags.isGeneratedAlways());
```